engine/eConstr.ml

```diff
@@ -651,6 +654,10 @@ let substnl subst n c = of_constr (Vars.substnl (List.map to_constr subst) n (to
 let substl subst c = of_constr (Vars.substl (List.map to_constr subst) (to_constr c))
 let subst1 c r = of_constr (Vars.subst1 (to_constr c) (to_constr r))
 
+let substnl_decl subst n d = of_rel_decl (Vars.substnl_decl (List.map to_constr subst) n (to_rel_decl d))
```

Comment:
OK, though I feel that the abstraction barrier would be clearer if put at the level of `EConstr` rather than at the level of the inner module `API`. 

But maybe it is also that I have to get used to this meaning of `unsafe`. Somehow, it does not mean "not to be used" or even "danger" but simply "may expose defined evars, be aware of it".